src/query/server/server.go

```diff
@@ -150,40 +248,9 @@ func Run(runOpts RunOptions) {
 		}
 	}
 
-	var clusters local.Clusters
-	if len(cfg.Clusters) > 0 {
-		opts := local.ClustersStaticConfigurationOptions{
-			AsyncSessions: true,
-		}
-		clusters, err = cfg.Clusters.NewClusters(opts)
-		if err != nil {
-			logger.Fatal("unable to connect to clusters", zap.Any("error", err))
-		}
-	} else {
-		localCfg := cfg.Local
-		if localCfg == nil {
-			localCfg = defaultLocalConfiguration
-		}
-		dbClientCh := runOpts.DBClient
-		if dbClientCh == nil {
-			logger.Fatal("no clusters configured and not running local cluster")
-		}
-		session := m3db.NewAsyncSession(func() (client.Client, error) {
-			return <-dbClientCh, nil
-		}, nil)
-		clusters, err = local.NewClusters(local.UnaggregatedClusterNamespaceDefinition{
-			NamespaceID: ident.StringID(localCfg.Namespace),
-			Session:     session,
-			Retention:   localCfg.Retention,
-		})
-		if err != nil {
-			logger.Fatal("unable to connect to clusters", zap.Any("error", err))
-		}
-	}
-
-	for _, namespace := range clusters.ClusterNamespaces() {
-		logger.Info("resolved cluster namespace",
-			zap.String("namespace", namespace.NamespaceID().String()))
+	clusters, err := initClusters(cfg, runOpts.DBClient, logger)
+	if err != nil {
+		return nil, nil, nil, nil, err
```

Comment:
nit: Wrap this with "unable to init clusters" or something?

Reply:
Oh I see you do this in the initClusters function; reads a little weird that you're wrapping one inside the function while wrapping the other one above; might be better to extract the err wrapping for initClusters and put it here

Reply:
there can be multiple conditions within initCluster for error so there's not a constant error wrapping I can use.